### PR TITLE
hub and spoke design, poc and library

### PR DESCRIPTION
**Pull request by nikitamarchenko**
This change is [Reviewable](https://reviewable.io/reviews/telstra/open-kilda/1706)

### CODE REVIEW

File: docs/design/hub-and-spoke/v7/poc/wfm/src/main/java/org/openkilda/hubandspoke/WorkerBolt.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            tasks.put(key, input);
+            processHubTuple(input);
+        } else if (sender.equals(componentSpoutWorker)) {
+            if (tasks.containsKey(key)) {
+                cancelCallback(key);
+                processAsyncResponseTuple(input);
+            } else {
+                log.error("missed task in worker bolt");
+            }
+        } else if (sender.equals(BOLT_COORDINATOR)) {
```
Comment (Settis) [inline docs/design/hub-and-spoke/v7/poc/wfm/src/main/java/org/openkilda/hubandspoke/WorkerBolt.java]: What about to move this check into CoordinatorClient?

File: docs/design/hub-and-spoke/v7/poc/wfm/src/main/java/org/openkilda/hubandspoke/HubBolt.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    @Override
+    public void execute(Tuple input) {
+        if (autoAck) {
+            collector.ack(input);
+        }
+        if (input.getSourceComponent().equals(componentSpoutHub)) {
+            processIncomeTuple(input);
+        } else if (input.getSourceComponent().equals(componentBoltWorker)) {
+            processWorkerResponseTuple(input);
+        } else if (input.getSourceComponent().equals(BOLT_COORDINATOR)) {
```
Comment (Settis) [inline docs/design/hub-and-spoke/v7/poc/wfm/src/main/java/org/openkilda/hubandspoke/HubBolt.java]: What about to move this check into CoordinatorClient?

File: docs/design/hub-and-spoke/v7/poc/wfm/src/main/java/org/openkilda/FlWorkerBolt.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
+package org.openkilda;
+
+import org.openkilda.hubandspoke.WorkerBolt;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.storm.tuple.Tuple;
+import org.apache.storm.tuple.Values;
+
+@Slf4j
+public class FlWorkerBolt extends WorkerBolt {
```
Comment (Settis) [inline docs/design/hub-and-spoke/v7/poc/wfm/src/main/java/org/openkilda/FlWorkerBolt.java]: Enjoyable look :)

File: docs/design/hub-and-spoke/v7/poc/wfm/src/main/java/org/openkilda/hubandspoke/WorkerBolt.java (hunk 1 above)
Comment (nikitamarchenko) [reply docs/design/hub-and-spoke/v7/poc/wfm/src/main/java/org/openkilda/hubandspoke/WorkerBolt.java]: more code more problems

**Comment by nikitamarchenko** [comment]
@Settis I am working on fsm solution right now.

**Comment by nikitamarchenko** [comment]
@carmine I will do that

File: docs/design/hub-and-spoke/v6/wfm-error-hub.puml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+sp_hub <<- bus
+sp_hub ->> bolt_hub : create H-tuple
+bolt_hub -> bolt_hub : prepare async jobs
+
+bolt_hub ->x bolt_hub: some error
+... hub spout timeout ...
+ack ->> sp_hub : fail
+alt retry
+    loop
+        sp_hub ->> bolt_hub: rerun all from start
```
[the remark this answers is not in the document]
Comment (nikitamarchenko) [reply docs/design/hub-and-spoke/v6/wfm-error-hub.puml]: done

File: docs/design/hub-and-spoke/v6/wfm.puml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+sp_work ->> bolt_work : create W-tuple
+== Async Operations ==
+bolt_work ->> bus
+async_job <<- bus
+async_job -> async_job
+async_job ->> bus
+== Handle responses ==
+sp_work <<- bus
+sp_work ->> bolt_work : create WR-tuple
+bolt_work ->> ack : ack on WR-tuple
```
[the remark this answers is not in the document]
Comment (nikitamarchenko) [reply docs/design/hub-and-spoke/v6/wfm.puml]: bc WR and W is a different tuple and not they don't anchored to each other.

**Comment by carmine** [comment]
LGTM

**Pull request merged**